recipe/meta.yaml

```
@@ -15,23 +24,45 @@ source:
     - 3599.patch
 
 build:
-  number: 2
+  number: {{ build }}
+
+  # add build string so packages can depend on
+  # mpi or nompi variants
+  # dependencies:
+  # `pkg * mpi_mpich_*` for mpich
+  # `pkg * mpi_*` for any mpi
+  # `pkg * nompi_*` for no mpi
+  {% if mpi == 'nompi' %}
+  {% set mpi_prefix = "nompi" %}
+  {% else %}
+  {% set mpi_prefix = "mpi_" + mpi %}
+  {% endif %}
+  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}
+
+  run_exports:
+    # strict runtime dependency on build-time MPI flavor
+    - {{ name }} * {{ mpi_prefix }}_*
+    # Releases are not (yet) compatible:
+    # There is no ABI compatibility check or guarantee between AMReX releases.
+    - amrex {{ version }}
```

Another way to express this is:
```suggestion
    # strict runtime dependency on build-time MPI flavor
    # Releases are not (yet) compatible:
    # There is no ABI compatibility check or guarantee between AMReX releases.
    - {{ pin_subpackage('amrex', min_pin='x.x.x', max_pin='x.x.x') }} {{ mpi_prefix }}_*
```
I am not sure what is preferred here.